src/ruby/puppetserver-lib/puppet/server/settings.rb

```diff
@@ -0,0 +1,85 @@
+require 'forwardable'
+
+class Puppet::Server::CompilerContext
+  THREAD_LOCAL_SETTINGS = Set.new([:strict, :code, :tasks])
+
+  def initialize(global_settings)
+    @settings = {}
+    THREAD_LOCAL_SETTINGS.each do |setting|
+      @settings[setting] = global_settings[setting]
+    end
+  end
+
+  def [](key)
+    @settings[key]
+  end
+end
+
+class Puppet::Server::Settings
+  extend Forwardable
+  THREAD_LOCAL_SETTINGS = Set.new([:strict, :code, :tasks])
+
+  def initialize(global_settings, puppet_config)
+    @global_settings = global_settings
+    @local_settings = Puppet::ThreadLocal.new {nil}
+    @puppet_config = puppet_config.reduce([]) do |acc, entry|
+      acc << "--#{entry[0]}"
+      acc << entry[1] if entry[1].kind_of?(String)
+      acc
+    end
```

Comment:
This feels very clojure-y to me.. I wonder if there's a more ruby-ish way to do this?

Also, could you replace `acc` with a full word, for clarity? I'm not sure what it's even meant to stand for.